scripts/feature_tests/upgrade/1cp_1w_bootDiskImage_cluster_upgrade.sh

```diff
@@ -0,0 +1,144 @@
+#!/bin/bash
+
+set -x
+
+# shellcheck disable=SC1091
+source ./common.sh
+
+echo '' >~/.ssh/known_hosts
+
+start_logging "${1}"
+
+set_number_of_master_node_replicas 1
+set_number_of_worker_node_replicas 1
+
+provision_controlplane_node
+wait_for_ctrlplane_provisioning_start
```

Comment:
this type of function can fail (provisioning not starting properly), In that case the cluster is deprovisioned within the function. However, the function returns normally, meaning that this script would continue with a deprovisioned cluster ? Maybe all functions doing similar error handling should cause an exit of the script, to prevent weird situations